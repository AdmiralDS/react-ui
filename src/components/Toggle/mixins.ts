import { css } from 'styled-components';

export type Dimension = 'm' | 's';
export type LabelPosition = 'left' | 'right';

const TOGGLE_WIDTH_M = '36px';
const TOGGLE_WIDTH_S = '28px';
const TOGGLE_HEIGHT_M = '20px';
const TOGGLE_HEIGHT_S = '16px';

export const SLIDER_SIZE_M = '16px';
export const SLIDER_SIZE_S = '12px';

export const sizes = css<{ dimension: Dimension }>`
  width: ${({ dimension }) => (dimension === 's' ? TOGGLE_WIDTH_S : TOGGLE_WIDTH_M)};
  height: ${({ dimension }) => (dimension === 's' ? TOGGLE_HEIGHT_S : TOGGLE_HEIGHT_M)};
`;

export const sliderSizes = css<{ dimension: Dimension }>`
  width: ${({ dimension }) => (dimension === 's' ? SLIDER_SIZE_S : SLIDER_SIZE_M)};
  height: ${({ dimension }) => (dimension === 's' ? SLIDER_SIZE_S : SLIDER_SIZE_M)};
  border-radius: ${({ dimension }) => (dimension === 's' ? SLIDER_SIZE_S : SLIDER_SIZE_M)};
`;
