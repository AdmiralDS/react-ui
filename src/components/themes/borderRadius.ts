import type { Shape } from '#src/components/themes/common';
export const ALL_BORDER_RADIUS_VALUES = [
  'Border radius 0',
  'Border radius 2',
  'Border radius 4',
  'Border radius 6',
  'Border radius 8',
  'Border radius 10',
] as const;

export function smallGroupBorderRadius(shape: Shape): string {
  switch (shape.borderRadiusKind) {
    case 'Border radius 0':
      return '0';
    case 'Border radius 2':
      return '2px';
    case 'Border radius 4':
      return '4px';
    case 'Border radius 6':
      return '4px';
    case 'Border radius 8':
      return '4px';
    case 'Border radius 10':
      return '4px';
    default:
      return '4px';
  }
}

export function mediumGroupBorderRadius(shape: Shape): string {
  switch (shape.borderRadiusKind) {
    case 'Border radius 0':
      return '0';
    case 'Border radius 2':
      return '2px';
    case 'Border radius 4':
      return '4px';
    case 'Border radius 6':
      return '6px';
    case 'Border radius 8':
      return '8px';
    case 'Border radius 10':
      return '10px';
    default:
      return '4px';
  }
}

export function largeGroupBorderRadius(shape: Shape): string {
  switch (shape.borderRadiusKind) {
    case 'Border radius 0':
      return '0';
    case 'Border radius 2':
      return '4px';
    case 'Border radius 4':
      return '8px';
    case 'Border radius 6':
      return '12px';
    case 'Border radius 8':
      return '16px';
    case 'Border radius 10':
      return '20px';
    default:
      return '8px';
  }
}
