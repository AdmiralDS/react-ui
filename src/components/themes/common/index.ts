import { Color } from '#/components/themes/common/color';
import { ZIndex } from '#/components/themes/common/zIndex';

export * from './color';

export interface Shadow {
  ClickableActive: string;
  ClickableDefault: string;
  ClickableHover: string;
  NonClickable: string;
}

export interface Theme {
  color: Color;
  shadow: Shadow;
  zIndex: ZIndex;
  name: 'light' | 'dark';
}
