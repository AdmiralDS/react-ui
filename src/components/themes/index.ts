export * from './light';
export * from './dark';
export type { Theme } from './common';
