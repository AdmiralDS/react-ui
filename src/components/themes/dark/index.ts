import type { Theme } from '../common';
import { COLOR as color } from './color';
import { shadow } from './shadow';
import { zIndex } from '../common/zIndex';

export const DARK_THEME: Theme = {
  color,
  shadow,
  zIndex,
  name: 'dark',
};
