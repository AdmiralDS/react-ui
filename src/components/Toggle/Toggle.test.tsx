import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent, { specialChars } from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components';

import { Toggle } from '../Toggle';
import { LIGHT_THEME } from '../themes';

describe('Toggle', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  const requiredProps = {
    onChange: jest.fn(),
    checked: false,
    'data-testid': 'element',
  };

  const Component = (props: any) => (
    <ThemeProvider theme={LIGHT_THEME}>
      <Toggle {...requiredProps} {...props}>
        text
      </Toggle>
    </ThemeProvider>
  );

  it('should render component with required properties', () => {
    const wrapper = render(<Component />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with disabled state', () => {
    const wrapper = render(<Component disabled />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with small dimension', () => {
    const wrapper = render(<Component dimension="s" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render component with left labelPosition', () => {
    const wrapper = render(<Component labelPosition="left" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange when user clicks on component', () => {
    const wrapper = render(<Component />);
    fireEvent.click(wrapper.getByLabelText('text'));
    const { onChange } = requiredProps;
    expect(onChange).toBeCalledTimes(1);
  });

  it('should focus input if user press Tab key', () => {
    render(<Component />);
    const [input] = screen.getAllByTestId('element');
    userEvent.tab();
    expect(input).toHaveFocus();
  });

  it('should call onChange if input is focused and user press Space key', () => {
    render(<Component />);
    const [input] = screen.getAllByTestId('element');
    userEvent.tab();
    userEvent.type(input, specialChars.space);
    const { onChange } = requiredProps;
    expect(onChange).toBeCalledTimes(2);
  });
});
