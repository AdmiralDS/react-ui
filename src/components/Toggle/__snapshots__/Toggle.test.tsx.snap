// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle should render component with disabled state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c1:checked + span {
  background: #0D69F2;
}

.c1:checked + span:before {
  left: calc(100% - 2px - 16px);
}

.c1:disabled + span {
  background: #A7B0BE;
}

.c1:checked:not(:disabled):hover + span,
.c1:checked:not(:disabled):focus + span {
  background: #0A54C2;
}

.c1:not(:checked):not(:disabled):hover + span,
.c1:not(:checked):not(:disabled):focus + span {
  background: #626F84;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 8px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #A7B0BE;
}

.c2 {
  position: relative;
  width: 36px;
  height: 20px;
  border-radius: 10px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: #8A96A8;
}

.c2:before {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 16px;
  height: 16px;
  border-radius: 16px;
  background: #FFFFFF;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: default;
  -webkit-tap-highlight-color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<body>
    <div>
      <label
        class="c0"
        disabled=""
        role="switch"
      >
        <input
          class="c1"
          data-testid="element"
          disabled=""
          type="checkbox"
        />
        <span
          aria-hidden="true"
          class="c2"
          disabled=""
        />
        <div
          class="c3"
          disabled=""
        >
          text
        </div>
      </label>
    </div>
  </body>,
  "container": .c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c1:checked + span {
  background: #0D69F2;
}

.c1:checked + span:before {
  left: calc(100% - 2px - 16px);
}

.c1:disabled + span {
  background: #A7B0BE;
}

.c1:checked:not(:disabled):hover + span,
.c1:checked:not(:disabled):focus + span {
  background: #0A54C2;
}

.c1:not(:checked):not(:disabled):hover + span,
.c1:not(:checked):not(:disabled):focus + span {
  background: #626F84;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 8px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #A7B0BE;
}

.c2 {
  position: relative;
  width: 36px;
  height: 20px;
  border-radius: 10px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: #8A96A8;
}

.c2:before {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 16px;
  height: 16px;
  border-radius: 16px;
  background: #FFFFFF;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: default;
  -webkit-tap-highlight-color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
    <label
      class="c0"
      disabled=""
      role="switch"
    >
      <input
        class="c1"
        data-testid="element"
        disabled=""
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="c2"
        disabled=""
      />
      <div
        class="c3"
        disabled=""
      >
        text
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Toggle should render component with left labelPosition 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c1:checked + span {
  background: #0D69F2;
}

.c1:checked + span:before {
  left: calc(100% - 2px - 16px);
}

.c1:disabled + span {
  background: #A7B0BE;
}

.c1:checked:not(:disabled):hover + span,
.c1:checked:not(:disabled):focus + span {
  background: #0A54C2;
}

.c1:not(:checked):not(:disabled):hover + span,
.c1:not(:checked):not(:disabled):focus + span {
  background: #626F84;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 8px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
}

.c2 {
  position: relative;
  width: 36px;
  height: 20px;
  border-radius: 10px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: #8A96A8;
}

.c2:before {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 16px;
  height: 16px;
  border-radius: 16px;
  background: #FFFFFF;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<body>
    <div>
      <label
        class="c0"
        role="switch"
      >
        <input
          class="c1"
          data-testid="element"
          type="checkbox"
        />
        <span
          aria-hidden="true"
          class="c2"
        />
        <div
          class="c3"
        >
          text
        </div>
      </label>
    </div>
  </body>,
  "container": .c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c1:checked + span {
  background: #0D69F2;
}

.c1:checked + span:before {
  left: calc(100% - 2px - 16px);
}

.c1:disabled + span {
  background: #A7B0BE;
}

.c1:checked:not(:disabled):hover + span,
.c1:checked:not(:disabled):focus + span {
  background: #0A54C2;
}

.c1:not(:checked):not(:disabled):hover + span,
.c1:not(:checked):not(:disabled):focus + span {
  background: #626F84;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-right: 8px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
}

.c2 {
  position: relative;
  width: 36px;
  height: 20px;
  border-radius: 10px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: #8A96A8;
}

.c2:before {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 16px;
  height: 16px;
  border-radius: 16px;
  background: #FFFFFF;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
    <label
      class="c0"
      role="switch"
    >
      <input
        class="c1"
        data-testid="element"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="c2"
      />
      <div
        class="c3"
      >
        text
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Toggle should render component with required properties 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c1:checked + span {
  background: #0D69F2;
}

.c1:checked + span:before {
  left: calc(100% - 2px - 16px);
}

.c1:disabled + span {
  background: #A7B0BE;
}

.c1:checked:not(:disabled):hover + span,
.c1:checked:not(:disabled):focus + span {
  background: #0A54C2;
}

.c1:not(:checked):not(:disabled):hover + span,
.c1:not(:checked):not(:disabled):focus + span {
  background: #626F84;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 8px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
}

.c2 {
  position: relative;
  width: 36px;
  height: 20px;
  border-radius: 10px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: #8A96A8;
}

.c2:before {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 16px;
  height: 16px;
  border-radius: 16px;
  background: #FFFFFF;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<body>
    <div>
      <label
        class="c0"
        role="switch"
      >
        <input
          class="c1"
          data-testid="element"
          type="checkbox"
        />
        <span
          aria-hidden="true"
          class="c2"
        />
        <div
          class="c3"
        >
          text
        </div>
      </label>
    </div>
  </body>,
  "container": .c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c1:checked + span {
  background: #0D69F2;
}

.c1:checked + span:before {
  left: calc(100% - 2px - 16px);
}

.c1:disabled + span {
  background: #A7B0BE;
}

.c1:checked:not(:disabled):hover + span,
.c1:checked:not(:disabled):focus + span {
  background: #0A54C2;
}

.c1:not(:checked):not(:disabled):hover + span,
.c1:not(:checked):not(:disabled):focus + span {
  background: #626F84;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 8px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 16px;
  line-height: 20px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
}

.c2 {
  position: relative;
  width: 36px;
  height: 20px;
  border-radius: 10px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: #8A96A8;
}

.c2:before {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 16px;
  height: 16px;
  border-radius: 16px;
  background: #FFFFFF;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
    <label
      class="c0"
      role="switch"
    >
      <input
        class="c1"
        data-testid="element"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="c2"
      />
      <div
        class="c3"
      >
        text
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Toggle should render component with small dimension 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c1:checked + span {
  background: #0D69F2;
}

.c1:checked + span:before {
  left: calc(100% - 2px - 12px);
}

.c1:disabled + span {
  background: #A7B0BE;
}

.c1:checked:not(:disabled):hover + span,
.c1:checked:not(:disabled):focus + span {
  background: #0A54C2;
}

.c1:not(:checked):not(:disabled):hover + span,
.c1:not(:checked):not(:disabled):focus + span {
  background: #626F84;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 8px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
}

.c2 {
  position: relative;
  width: 28px;
  height: 16px;
  border-radius: 10px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: #8A96A8;
}

.c2:before {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 12px;
  height: 12px;
  border-radius: 12px;
  background: #FFFFFF;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<body>
    <div>
      <label
        class="c0"
        role="switch"
      >
        <input
          class="c1"
          data-testid="element"
          type="checkbox"
        />
        <span
          aria-hidden="true"
          class="c2"
        />
        <div
          class="c3"
        >
          text
        </div>
      </label>
    </div>
  </body>,
  "container": .c1 {
  position: absolute;
  width: 0;
  height: 0;
  opacity: 0;
}

.c1:checked + span {
  background: #0D69F2;
}

.c1:checked + span:before {
  left: calc(100% - 2px - 12px);
}

.c1:disabled + span {
  background: #A7B0BE;
}

.c1:checked:not(:disabled):hover + span,
.c1:checked:not(:disabled):focus + span {
  background: #0A54C2;
}

.c1:not(:checked):not(:disabled):hover + span,
.c1:not(:checked):not(:disabled):focus + span {
  background: #626F84;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-left: 8px;
  font-family: 'VTB Group UI',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 16px;
  font-feature-settings: 'tnum' on,'lnum' on;
  color: #2B313B;
  color: #2B313B;
}

.c2 {
  position: relative;
  width: 28px;
  height: 16px;
  border-radius: 10px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  background: #8A96A8;
}

.c2:before {
  content: '';
  position: absolute;
  top: 2px;
  left: 2px;
  width: 12px;
  height: 12px;
  border-radius: 12px;
  background: #FFFFFF;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

<div>
    <label
      class="c0"
      role="switch"
    >
      <input
        class="c1"
        data-testid="element"
        type="checkbox"
      />
      <span
        aria-hidden="true"
        class="c2"
      />
      <div
        class="c3"
      >
        text
      </div>
    </label>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
